proto/v1/activity_service.proto

```diff
@@ -0,0 +1,63 @@
+syntax = "proto3";
+
+package bytebase.v1.activity;
+
+option go_package = "generated-go/v1";
+
+message Activity {
+  enum Type {
+    TYPE_UNSPECIFIED = 0;
+    // Issue related activity types.
+    //
+    // TYPE_ISSUE_CREATE represents creating an issue.
+    TYPE_ISSUE_CREATE = 1;
+    // TYPE_ISSUE_COMMENT_CREATE represents commenting on an issue.
+    TYPE_ISSUE_COMMENT_CREATE = 2;
+    // TYPE_ISSUE_FIELD_UPDATE represents updating the issue field, likes title, description, assignee, etc.
+    TYPE_ISSUE_FIELD_UPDATE = 3;
+    // TYPE_ISSUE_STATUS_UPDATE represents the issue status change, including OPEN, CLOSE, CANCEL fow now.
+    TYPE_ISSUE_STATUS_UPDATE = 4;
+    // TYPE_ISSUE_PIPELINE_STAGE_STATUS_UPDATE represents the pipeline stage status change, including BEGIN, END for now.
+    TYPE_ISSUE_PIPELINE_STAGE_STATUS_UPDATE = 5;
+    // TYPE_ISSUE_PIPELINE_TASK_STATUS_UPDATE represents the pipeline task status change, including PENDING, PENDING_APPROVAL, RUNNING, SUCCESS, FAILURE, CANCELED for now.
+    TYPE_ISSUE_PIPELINE_TASK_STATUS_UPDATE = 6;
+    // TYPE_ISSUE_PIPELINE_TASK_FILE_COMMIT represents the VCS trigger to commit a file to update the task statement.
+    TYPE_ISSUE_PIPELINE_TASK_FILE_COMMIT = 7;
+    // TYPE_ISSUE_PIPELINE_TASK_STATEMENT_UPDATE represents the manual update of the task statement.
+    TYPE_ISSUE_PIPELINE_TASK_STATEMENT_UPDATE = 8;
+    // TYPE_ISSUE_PIPELINE_TASK_EARLIEST_ALLOWED_TIME_UPDATE represents the manual update of the task earliest allowed time.
+    TYPE_ISSUE_PIPELINE_TASK_EARLIEST_ALLOWED_TIME_UPDATE = 9;
+
+    // Member related activity types.
+    //
+    // TYPE_MEMBER_CREATE represents creating a members.
+    TYPE_MEMBER_CREATE = 10;
+    // TYPE_MEMBER_ROLE_UPDATE represents updating the member role, for example, from ADMIN to MEMBER.
+    TYPE_MEMBER_ROLE_UPDATE = 11;
+    // TYPE_MEMBER_ACTIVATE represents activating a deactivated member.
+    TYPE_MEMBER_ACTIVATE = 12;
+    // TYPE_MEMBER_DEACTIVATE represents deactivating an active member.
+    TYPE_MEMBER_DEACTIVATE = 13;
+
+    // Project related activity types.
+    //
+    // TYPE_PROJECT_REPOSITORY_PUSH represents Bytebase receiving a push event from the project repository.
+    TYPE_PROJECT_REPOSITORY_PUSH = 14;
+    // TYPE_PROJECT_DATABASE_TRANFER represents transfering the database from one project to another.
+    TYPE_PROJECT_DATABASE_TRANSFER = 15;
+    // TYPE_PROJECT_MEMBER_CREATE represents adding a member to the project.
+    TYPE_PROJECT_MEMBER_CREATE = 16;
```

Comment:
Nit: good for now, we want to merge some of these cases later.

Reply:
Let's modify it when we modify the activity.